src/Services/InventoryUpdateService.php

```diff
@@ -37,283 +50,360 @@ class InventoryUpdateService
 
-        $amtOfDtosForPage = count($validatedRequestDTOs);
+        $amtOfDtosForPage = count($requestDTOsForPage);
 
         if ($amtOfDtosForPage <= 0) {
-          $loggerContract
+          $this->logger
             ->debug(
               TranslationHelper::getLoggerKey(self::LOG_KEY_DEBUG),
               [
-                'additionalInfo' => ['info' => 'No items to update'],
+                'additionalInfo' => [
+                  'full' => (string) $fullInventory,
+                  'pageNum' => $pageNumber,
+                  'info' => 'No request DTOs to send'
+                ],
                 'method' => __METHOD__
               ]
             );
 
-          $externalLogs->addInfoLog('Inventory ' . ($fullInventory ? 'Full' : '') . ': No items to update');
+          $externalLogs->addInfoLog('Inventory ' . ($fullInventory ? 'Full' : '') . ': No items to update for page ' . $pageNumber);
         } else {
-          $externalLogs->addInfoLog('Inventory ' . ($fullInventory ? 'Full' : '') . ': ' . (string) $amtOfDtosForPage . ' updates to send');
+          $amtOfVariationsForPage = count($variationIdsForPage);
+          $totalVariationIdsInDTOsForAllPages += $amtOfVariationsForPage;
 
-          $loggerContract->debug(
+          $externalLogs->addInfoLog('Inventory ' . ($fullInventory ? 'Full' : '') . ': ' . (string) $amtOfDtosForPage . ' updates to send for page ' . $pageNumber);
+
+          $this->logger->debug(
             TranslationHelper::getLoggerKey(self::LOG_KEY_DEBUG),
             [
-              'additionalInfo' => ['info' => (string) $amtOfDtosForPage . ' updates to send'],
+              'additionalInfo' => [
+                'full' => (string) $fullInventory,
+                'page' => $pageNumber,
+                'amtOfDtosForPage' => $amtOfDtosForPage,
+                'amtOfVariationsForPage' => $amtOfVariationsForPage
+              ],
               'method' => __METHOD__
             ]
           );
 
-          $saveInventoryDuration += TimeHelper::getMilliseconds() - $msAtPageStart;
-          $msBeforeUpdate = TimeHelper::getMilliseconds();
+          $totalTimeSpentGatheringData += TimeHelper::getMilliseconds() - $unixTimeAtPageStart;
+          $unixTimeBeforeSendingData = TimeHelper::getMilliseconds();
 
-          $inventorySaveTotal +=  $amtOfDtosForPage;
+          $totalDtosAttempted +=  $amtOfDtosForPage;
 
-          $dto = $inventoryService->updateBulk($validatedRequestDTOs, $fullInventory);
+          $responseDto = $this->inventoryService->updateBulk($requestDTOsForPage, $fullInventory);
 
-          $savedInventoryDuration += TimeHelper::getMilliseconds() - $msBeforeUpdate;
+          $totalTimeSpentSendingData += TimeHelper::getMilliseconds() - $unixTimeBeforeSendingData;
 
-          $inventorySaveSuccess += count($validatedRequestDTOs) - count($dto->getErrors());
-          $inventorySaveFail += count($dto->getErrors());
+          $amtErrors = count($responseDto->getErrors());
```

Comment:
Can we store the detail errors `$responseDto->getErrors()` and send together with `totalDtosFailed` to external log as well?

Reply:
great idea.

Reply:
due to memory constraints, I'm logging them out immediately instead of queuing.

Reply:
I still don't see the logging for this. Am I missing something?

Reply:
yeah check out around line 260:
```
 $amtErrors = 0;
          $errors = $responseDto->getErrors();

          if (isset($errors) && !empty($errors))
          {
            $amtErrors = count($errors);

            $this->logger->error(TranslationHelper::getLoggerKey(self::LOG_KEY_INVENTORY_ERRORS), [
              'additionalInfo' => [
                'full' => (string) $fullInventory,
                'errors' => json_encode($errors)
              ],
              'method' => __METHOD__
            ]);

            $externalLogs->addErrorLog('Inventory ' . ($fullInventory ? 'Full' : '') . ' errors from page ' . $pageNumber . ': ' . json_encode($errors));
```

Reply:
aha, saw it now, somehow git cache the previous file.